src/plans/D3MAaveDaiPlan.sol

```diff
@@ -0,0 +1,213 @@
+        emit File(what, data);
+    }
+
+    function _maxBar() internal view returns (uint256) {
+        return interestStrategy.getMaxVariableBorrowRate();
+    }
+
+    // --- Automated Rate targeting ---
+    function calculateTargetSupply(uint256 targetInterestRate) external view returns (uint256) {
+        uint256 totalDebt = _add(stableDebt.totalSupply(), variableDebt.totalSupply());
+        return _calculateTargetSupply(targetInterestRate, totalDebt);
+    }
+
+    function _calculateTargetSupply(uint256 targetInterestRate, uint256 totalDebt) internal view returns (uint256) {
+        uint256 base = interestStrategy.baseVariableBorrowRate();
+        require(targetInterestRate > base, "D3MAaveDaiPlan/target-interest-base");
+        require(targetInterestRate <= _maxBar(), "D3MAaveDaiPlan/above-max-interest");
+
+        // Do inverse calculation of interestStrategy
+        uint256 variableRateSlope1 = interestStrategy.variableRateSlope1();
+        uint256 targetUtil;
+        if (targetInterestRate > _add(base, variableRateSlope1)) {
+            // Excess interest rate
+            uint256 r = targetInterestRate - base - variableRateSlope1;
+            targetUtil = _add(
+                            _rdiv(
+                                _rmul(interestStrategy.EXCESS_UTILIZATION_RATE(), r),
+                                interestStrategy.variableRateSlope2()
+                            ),
+                            interestStrategy.OPTIMAL_UTILIZATION_RATE()
+                        );
+        } else {
+            // Optimal interest rate
+            targetUtil = _rdiv(_rmul(_sub(targetInterestRate, base), interestStrategy.OPTIMAL_UTILIZATION_RATE()), variableRateSlope1);
+        }
+        return _rdiv(totalDebt, targetUtil);
+    }
+
+    function getTargetAssets(uint256 currentAssets) external override view returns (uint256) {
+        uint256 targetInterestRate = bar;
+        if (targetInterestRate == 0) return 0;  // De-activated
+
+        uint256 totalDebt = _add(stableDebt.totalSupply(), variableDebt.totalSupply());
+
+        uint256 totalPoolSize = _add(
+                TokenLike(dai).balanceOf(adai),
+                totalDebt
+            );
+
+        uint256 targetTotalPoolSize = _calculateTargetSupply(targetInterestRate, totalDebt);
+        if (targetTotalPoolSize >= totalPoolSize) {
+            // Increase debt (or same)
+            return _add(currentAssets, targetTotalPoolSize - totalPoolSize);
+        } else {
+            // Decrease debt
+            uint256 decrease = totalPoolSize - targetTotalPoolSize;
+            if (currentAssets >= decrease) {
+                return currentAssets - decrease;
+            } else {
+                return 0;
+            }
+        }
+    }
+
+    function active() public view override returns (bool) {
+        (,,,,,,,,,, address strategy,) = pool.getReserveData(address(dai));
+        return strategy == address(interestStrategy);
```

Comment:
For aave, consider also checking that `bar != 0`

Reply:
Consider if we want an implementation check on the aave pool proxy either here or in disable.

Reply:
> Consider if we want an implementation check on the aave pool proxy either here or in disable.

@brianmcmichael or @godsflaw I'm not sure i remember what we meant by checking the aave pool proxy. Looking at the pool, it looks like the `implementation()` view function requires the caller to be an admin: https://etherscan.io/address/0x7d2768dE32b0b80b7a3454c06BdAc94A69DDc7A9#code